department-of-veterans-affairs/caseflow
# Caseflow 3383 add homelessness to intake review screen

Resolves [CASEFLOW-3383](https://vajira.max.gov/browse/CASEFLOW-3383)

### Description
If merged, these changes adds a new homelessness radio button group for homelessness on the Caseflow intake 10182 form.

### Acceptance Criteria


- [x]  Add a new radio button option for homelessness to the intake form.
- [x] Potential selections are "Yes" and "No".
- [x]  The selection itself is optional, meaning the user should be able to progress in the Intake process without making a selection.
- [x]  As far as the code is concerned, the lack of a selection should be considered a "No" response.
- [x] Selection of an option needs to be managed and be accessible via the global Redux state.
- [x] The field we should be populating in the database with the selection once the form is submitted is the homelessness column on the appeals table which was completed in [CASEFLOW-3397](https://vajira.max.gov/browse/CASEFLOW-3397).

### Testing Plan
1. Ensure you have enabled the `updated_appeal_form` FeatureToggle. Load the rails console (`make c`/`bundle exec rails console`) and enter `FeatureToggle.enable!(:updated_appeal_form)` to enable it.
2. Navigate to Caseflow Intake (`/intake`) and begin the Form 10182 entry flow.
3. The final field on the Review Form page will be the homelessness prompt. Select Yes, No, or don't select any option (this will default to No/False), and continue through the 10182 entry process.
4. Upon submitting your form, you can check the entry for the appeal you've submitted in the 
[appeals table](https://department-of-veterans-affairs.github.io/caseflow/task_trees/schema/caseflow_schema_csv.html#appeals) in the Caseflow database to see if the `homelessness` column's value matches what you've entered.


### User Facing Changes

 BEFORE|AFTER
---|---
 ![image](https://user-images.githubusercontent.com/22192993/159284245-98bcd887-d985-49c9-b1a0-d3d047b564e2.png)
![image](https://user-images.githubusercontent.com/22192993/159284724-03effb88-913e-472c-a9d9-56613940a9c9.png)
